### PR TITLE
Fix deprecations and re-enable Werror

### PR DESCRIPTION
**Pull request by hunterwerlla**
## Description
- Re-add `kotlinOptions.allWarningsAsErrors = true`
- Delete telemetry namespace

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/SamProjectWizard.kt [hunk 1]
```diff
@@ -147,8 +147,8 @@ abstract class SamProjectTemplate {
             hasException = true
             throw e
         } finally {
-            TelemetryService.getInstance().record(project, "SAM") {
-                datum("Init") {
+            TelemetryService.getInstance().record(project) {
+                datum("SAM_Init") {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/SamProjectWizard.kt]: Should be `SAM.Init`

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/DeployServerlessApplicationDialog.kt [hunk 2]
```diff
@@ -32,7 +32,7 @@ import javax.swing.JComponent
 class DeployServerlessApplicationDialog(
     private val project: Project,
     private val templateFile: VirtualFile
-) : LoggingDialogWrapper(project) {
+) : DialogWrapper(project) {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/DeployServerlessApplicationDialog.kt]: Do we need telemetry for these dialogs?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamInvokeRunner.kt [hunk 3]
```diff
@@ -118,32 +117,26 @@ class SamInvokeRunner : AsyncProgramRunner<RunnerSettings>() {
...
-                    .whenComplete { account, _ ->
-                        TelemetryService.getInstance().record(
-                            "SamInvoke",
-                            TelemetryService.MetricEventMetadata(
-                                awsAccount = account ?: METADATA_INVALID,
-                                awsRegion = lambdaSettings.region.id
-                            )
-                        ) {
+                    .whenComplete { _, _ ->
+                        TelemetryService.getInstance().record(state.environment.project) {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamInvokeRunner.kt]: Should this be the Lambda account/region or the IDE account/region

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/SamProjectWizard.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/SamProjectWizard.kt]: Fixed

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/DeployServerlessApplicationDialog.kt (hunk 2 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/DeployServerlessApplicationDialog.kt]: The telemetry for this was pretty useless (it was DeployServerlessApplicationDialog.OkAction / DeployServerlessApplicationDialog.CancelAction). Aded telemetry in `DeployServerlesssApplicationAction` where it was used.

Will take a look at more uses of this tomorrow

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamInvokeRunner.kt (hunk 3 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamInvokeRunner.kt]: true, reverted change

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployDialog.kt [hunk 4]
```diff
@@ -214,8 +214,8 @@ class SamDeployDialog(
         }
 
         return future.whenComplete { _, exception ->
-            TelemetryService.getInstance().record(project, "SamDeploy") {
-                datum(title) {
+            TelemetryService.getInstance().record(project) {
+                datum("SamDeploy_$title") {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployDialog.kt]: missed this

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamInvokeRunner.kt [hunk 5]
```diff
@@ -120,30 +120,29 @@ class SamInvokeRunner : AsyncProgramRunner<RunnerSettings>() {
...
                         TelemetryService.getInstance().record(
-                            "SamInvoke",
                             TelemetryService.MetricEventMetadata(
                                 awsAccount = account ?: METADATA_INVALID,
                                 awsRegion = lambdaSettings.region.id
                             )
                         ) {
                             val type = if (environment.isDebug()) "Debug" else "Run"
-                            datum(type) {
+                            datum("SamInvoke_$type") {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamInvokeRunner.kt]: and this one

Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamInvokeRunner.kt]: fixed

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployDialog.kt (hunk 4 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployDialog.kt]: fixed

**Pull request merged**